### ISSUE TITLE
CA2241: Provide correct arguments to formatting methods fires on IFormattable.ToString

### ISSUE THREAD

**Issue by manfred-brands** [dotnet/roslyn-analyzers#7023]
### Analyzer

**Diagnostic ID**: [CA2241](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2241): `Provide correct arguments to formatting methods`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 8.0.0-preview.23472.1 (Latest)

### Describe the bug

`IFormattable.ToString(string? format, IFormatProvider? provider)` is treated as if it is a format specification.

### Steps To Reproduce

Any code where to `IFormattable.ToString` is used *more than once*:

```csharp
public string FormatLargest(double value1, double value2)
{
    return value1 >= value2 ?
        value1.ToString("F1", CultureInfo.InvariantCulture) :
        value2.ToString("F1", CultureInfo.InvariantCulture);
}
```

### Expected behavior

CA2241 is not flagged.

### Actual behavior

![image](https://github.com/dotnet/roslyn-analyzers/assets/8655218/fcda6361-c47a-4fca-b3ec-6548c420df8a)

Note that the 1st usage is not flagged.

## Additional context

Introduced, unfortunately by me, with #3653 
Only shows up when: `try_determine_additional_string_formatting_methods_automatically = true`

I will create a PR to fix this shortly.
